pkgs/development/interpreters/octave/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, gfortran, readline, ncurses, perl, flex, texinfo, qhull
```

Comment:
Texinfo is not required at build time because the source archive already has compiled versions of all the documentation (and also other machine-independent stuff, like scripts).  This can be seen by comparing the contents of `doc/interpreter` for the existing octave package and the proposed octaveHg one, e.g.,

~~~ bash
nix-shell -A octave '<nixpkgs>'
unpackPhase
ls octave-4.2.1/doc/interpreter/
~~~

For the `octave` package, you will see all the info, PDF, etc. files there already.

However, I think I've made an mistake, as texinfo *is* required, but at runtime rather than build time, as the `makeinfo` command is used by `help`.  Actually, this seems to be broken with current normal Octave already:

~~~
$ nix-shell --pure -p octave --run "octave --eval 'help sparse'" '<nixpkgs>'
sh: makeinfo: command not found
warning: help: Texinfo formatting filter exited abnormally; raw Texinfo source of help text follows...
'sparse' is a built-in function from the file libinterp/corefcn/sparse.cc



Additional help for built-in functions and operators is
available in the online version of the manual.  Use the command
'doc <topic>' to search the manual index.

Help and information about Octave is also available on the WWW
at http://www.octave.org and via the help@octave.org
mailing list.
~~~

`nix-shell --pure -p octave texinfo --run "octave --eval 'help sparse'" '<nixpkgs>'` works though. 

I'll rework this PR to make texinfo a runtime dep of Octave.